src/js/components/Search.js

```diff
@@ -180,20 +191,42 @@ export default class Search extends Component {
     this.setState({activeSuggestionIndex: index});
   }
 
-  _onEnter () {
+  _onEnter (event) {
+    event.preventDefault(); // prevent submitting forms
     this._onRemoveDrop();
     if (this.state.activeSuggestionIndex >= 0) {
       var suggestion = this.props.suggestions[this.state.activeSuggestionIndex];
+      this.setState({value: suggestion});
       if (this.props.onChange) {
-        this.props.onChange(suggestion);
+        this.props.onChange(suggestion, true);
+      }
+      if (this.props.onSelect) {
+        this.props.onSelect({
+          target: this.refs.input || this.refs.control,
+          suggestion: suggestion
+        }, true);
+      }
+    } else {
+      if (this.props.onSelect) {
+        this.props.onSelect({
+          target: this.refs.input || this.refs.control,
+          suggestion: suggestion
```

Comment:
The 'suggestion' variable is a bit fragile here. I think we should initialize it after line 196.